hudi-common/src/main/java/org/apache/hudi/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -90,15 +101,25 @@ public void withOutputStream(FSDataOutputStream output) {
    */
   private FSDataOutputStream getOutputStream() throws IOException {
     if (this.output == null) {
-      Path path = logFile.getPath();
-      if (fs.exists(path)) {
-        rollOver();
-        createNewFile();
-        LOG.info("File {} already exists, rolling over to {}", path, logFile.getPath());
-      } else {
-        LOG.info(logFile + " does not exist. Create a new file");
-        // Block size does not matter as we will always manually autoflush
-        createNewFile();
+      boolean created = false;
+      while (!created) {
+        try {
+          // Block size does not matter as we will always manually autoflush
+          createNewFile();
+          LOG.info("Created a new log file: {}", logFile);
+          created = true;
+        } catch (FileAlreadyExistsException ignored) {
+          LOG.info("File {} already exists, rolling over", logFile.getPath());
+          rollOver();
+        } catch (RemoteException re) {
+          if (re.getClassName().contentEquals(AlreadyBeingCreatedException.class.getName())) {
+            LOG.warn("Another task executor writing to the same log file(" + logFile + ", rolling over");
+            // Rollover the current log file (since cannot get a stream handle) and create new one
+            rollOver();
+          } else {
+            throw re;
```

Comment:
When will it come to ths else block? Also, please throw a HoodieException with some message.

Reply:
Sorry, I don't think we need a wrapper `HoodieException`